javac-plugin/src/main/java/org/moditect/deptective/internal/graph/GraphUtils.java

```diff
@@ -0,0 +1,177 @@
+/**
+ *  Copyright 2019 The ModiTect authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.moditect.deptective.internal.graph;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.moditect.deptective.internal.graph.impl.DependencyStructureMatrix;
+import org.moditect.deptective.internal.graph.impl.FastFasSorter;
+import org.moditect.deptective.internal.graph.impl.Tarjan;
+
+/**
+ * 
+ * @author Gerd W&uuml;therich (gw@code-kontor.io)
+ */
+public class GraphUtils {
+
+	/**
+	 * A directed graph is called strongly connected if there is a path in each direction between each pair of vertices of the graph.
+	 * A strongly connected component (SCC) of a directed graph is a maximal strongly connected subgraph. 
+	 * 
+	 * @param nodes the collection of nodes (the directed graph)
+	 * @return a list of strongly connected components (SCCs). Note that the result also contains components that contain only a single node.
+	 */
+	public static <N extends INode> List<List<N>> detectStronglyConnectedComponents(Collection<? extends N> nodes) {
+
+		return new Tarjan<N>().detectStronglyConnectedComponents(checkNotNull(nodes));
+	}
+
+	/**
+	 * 
+	 * @param nodes
+	 * @return
+	 */
+	public static <N extends INode> List<List<N>> detectCycles(Collection<N> nodes) {
+		return new Tarjan<N>().detectStronglyConnectedComponents(nodes).stream().filter(cycle -> cycle.size() > 1)
+				.collect(Collectors.toList());
+	}
+
+	/**
+	 * 
+	 * @param nodes
+	 * @return
+	 */
+	public static <N extends INode, D extends IDependency> IDependencyStructureMatrix<N, D> createDependencyStructureMatrix(Collection<N> nodes) {
+		return new DependencyStructureMatrix<N, D>(nodes);
+	}
+
+	/**
```

Comment:
Could you please get rid of all the empty comments? Let's only docs if they are actually meaningful.